src/pocketmine/level/Location.php

```diff
@@ -57,6 +57,15 @@ public static function fromObject(Vector3 $pos, Level $level = null, $yaw = 0.0,
 		return new Location($pos->x, $pos->y, $pos->z, $yaw, $pitch, $level ?? (($pos instanceof Position) ? $pos->level : null));
 	}
 
+	/**
+	 * Return a Location instance
+	 * 
+	 * @return Location
+	 */
+	public function asLocation() : Location {
+		return new Location($this->x, $this->y, $this->z, $this->yaw, $this->pitch, $this->level);
```

Comment:
That would return descendents of the class, which defeats the object of doing this. (No pun intended).